src/Aspire.Hosting/ApplicationModel/EndpointAnnotation.cs

```diff
@@ -30,8 +27,8 @@ public sealed class EndpointAnnotation : IResourceAnnotation
     /// <param name="port">Desired port for the service.</param>
     /// <param name="targetPort">This is the port the resource is listening on. If the endpoint is used for the container, it is the container port.</param>
     /// <param name="isExternal">Indicates that this endpoint should be exposed externally at publish time.</param>
-    /// <param name="isProxied">Specifies if the endpoint will be proxied by DCP. Defaults to true.</param>
-    public EndpointAnnotation(ProtocolType protocol, string? uriScheme = null, string? transport = null, [EndpointName] string? name = null, int? port = null, int? targetPort = null, bool? isExternal = null, bool isProxied = true)
+    /// <param name="isProxied">Specifies if the endpoint will be proxied by DCP. Defaults to null.</param>
+    public EndpointAnnotation(ProtocolType protocol, string? uriScheme = null, string? transport = null, [EndpointName] string? name = null, int? port = null, int? targetPort = null, bool? isExternal = null, bool? isProxied = null)
```

Comment:
We could go with a hybrid approach and still force persistent endpoints to be un-proxied if we don't want these API changes.

Reply:
Reverted the potential API changes.